src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c

```diff
@@ -27,15 +31,148 @@ const ActorInit Obj_Ice_Poly_InitVars = {
+    this->colliderIce.dim.radius *= thisx->scale.x;
+    this->colliderIce.dim.height *= thisx->scale.y;
+    this->colliderHard.dim.radius *= thisx->scale.x;
+    this->colliderHard.dim.height *= thisx->scale.y;
+    Actor_SetHeight(thisx, thisx->scale.y * 30.0f);
+    this->actionFunc = ObjIcePoly_Idle;
+}
+
+void ObjIcePoly_Destroy(Actor* thisx, GlobalContext *globalCtx) {
+    s32 pad;
+    ObjIcePoly* this = THIS;
+
+    if ((this->actor.params >= 0) && (this->actor.params < 3)) {
+        Collider_DestroyCylinder(globalCtx, &this->colliderIce);
+        Collider_DestroyCylinder(globalCtx, &this->colliderHard);
+    }
+}
+
+void ObjIcePoly_Idle(ObjIcePoly *this, GlobalContext *globalCtx) {
+    static Vec3f zeroVec = {0.0f, 0.0f, 0.0f};
+    s32 pad;
+    Vec3f pos;
+
+    if (this->colliderIce.base.acFlags & 2) {
+        this->meltTimer = -this->colliderIce.body.acHitItem->toucher.damage;
+        this->actor.posRot2.rot.y = this->actor.yawTowardsLink;
+        func_800800F8(globalCtx, 0x1400, 40, &this->actor, 0);
+        this->actionFunc = ObjIcePoly_Melt;
+    } else if (this->actor.parent != NULL) {
+        this->actor.parent->freezeTimer = 40;
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->colliderIce.base);
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colliderIce.base);
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->colliderIce.base);
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colliderHard.base);
+    } else {
+        Actor_Kill(&this->actor);
+    }
+    pos.x = this->actor.posRot.pos.x + this->actor.scale.x * (Math_Rand_S16Offset(15, 15) * (Math_Rand_ZeroOne() < 0.5f ? -1 : 1));
+    pos.y = this->actor.posRot.pos.y + this->actor.scale.y * Math_Rand_S16Offset(10, 90);
+    pos.z = this->actor.posRot.pos.z + this->actor.scale.z * (Math_Rand_S16Offset(15, 15) * (Math_Rand_ZeroOne() < 0.5f ? -1 : 1));
+    if ((globalCtx->gameplayFrames % 7) == 0) {
+        EffectSsKiraKira_SpawnDispersed(globalCtx, &pos, &zeroVec, &zeroVec, &sColorWhite, &sColorGray, 0x7D0, 5);
+    }
+}
+
+void ObjIcePoly_Melt(ObjIcePoly *this, GlobalContext *globalCtx) {
+    Vec3f accel;
+    Vec3f vel;
+    Vec3f pos;
+    s32 i;
+
+    accel.x = 0.0f;
+    accel.y = this->actor.scale.y;
+    accel.z = 0.0f;
+    vel.x = 0.0f;
+    vel.y = this->actor.scale.y;
+    vel.z = 0.0f;
+
+    for(i = 0; i < 2; i++) {
+        pos.x = this->actor.posRot.pos.x + this->actor.scale.x * (Math_Rand_S16Offset(20, 20) * (Math_Rand_ZeroOne() < 0.5f ? -1 : 1));
+        pos.y = this->actor.posRot.pos.y + this->actor.scale.y * Math_Rand_ZeroOne() * 50.0f;
+        pos.z = this->actor.posRot.pos.z + this->actor.scale.x * (Math_Rand_S16Offset(20, 20) * (Math_Rand_ZeroOne() < 0.5f ? -1 : 1));
+        func_8002829C(globalCtx, &pos, &vel, &accel, &sColorWhite, &sColorGray, Math_Rand_S16Offset(0x15E, 0x64) * this->actor.scale.x, this->actor.scale.x * 20.0f);
```

Comment:
was the formatter run? this line is pretty long, Id think it would snap that down

Reply:
Apparently I forgot to. Formatted.